pandas/io/parquet.py

```diff
@@ -113,9 +113,22 @@ def write(self, df, path, compression='snappy',
             table = self.api.Table.from_pandas(df, timestamps_to_ms=True)
             self.api.parquet.write_table(
                 table, path, compression=compression, **kwargs)
-
         else:
             table = self.api.Table.from_pandas(df)
+            custom_metadata = kwargs.pop('metadata', {})
+            if custom_metadata:
+                if 'pandas' in custom_metadata:
+                    warn(
```

Comment:
There should also be a test that asserts this warning actually happens

Reply:
OK